static/js/ReaderPanel.jsx

```diff
@@ -1382,7 +1383,15 @@ class ReaderControls extends Component {
           />
           <DisplaySettingsButton onClick={this.props.openDisplaySettings} />
         </div>);
-    let transLangPrefSuggBann = hideHeader || connectionsHeader ? null : <TranslationLanguagePreferenceSuggestionBanner setTranslationLanguagePreference={this.props.setTranslationLanguagePreference} />;
+    const openTranslations = this.props.openConnectionsPanel.bind(null, [this.props.currentRef], null, {"connectionsMode": "Translations"});
+    const transCallToActionApplies = () => Sefaria.transCallToActionApplies(this.props.currentBook(), this.props.settings.language);
+    let banner = (hideHeader || connectionsHeader) ? null : (
+        <TextColumnBannerChooser
+            setTranslationLanguagePreference={this.props.setTranslationLanguagePreference}
+            openTranslations={openTranslations}
+            transCallToActionApplies={transCallToActionApplies}
```

Comment:
Its more props to pass, but I's prefer passing in the book and language rather than passing in the callback, it seems less dependent that way to me?

Reply:
It is actually less coupled to the ReaderControls component when you pass a callback. I think it's likely the business logic of when it will apply will require different pieces of data and it felt less random than passing seemingly unrelated props (current book and language). But if you still think it's better, happy to change it. It's not clear to me my way is better, but I wasn't sure if you understood my considerations.

Reply:
I understand. My main issue is that we now have a function called "transCallToActionApplies" sitting completely out of context from the one component its really relevant to. Maybe renaming it?
Also, if it stays here it should also be moved out of the render? Though I can understand more why its there, since its technically part of the render, but still.